src/Gui/MainWindow.cpp

```diff
@@ -1665,6 +1666,11 @@ void MainWindow::_updateActions()
         Application::Instance->commandManager().testActive();
     }
     d->actionUpdateDelay = 0;
+    QMdiSubWindow* mdi = d->mdiArea->currentSubWindow();
+    if (mdi) {
+        auto view = dynamic_cast<MDIView*>(mdi->widget());
```

Comment:
This could be a good helper, something like `getCurrentWindow`/`getCurrentView` then this function would not care about how to access current view.